inst/app/modules/main_map.R

```diff
@@ -0,0 +1,51 @@
+
+hk_accidents_valid <- dplyr::filter(hk_accidents, !is.na(latitude) & !is.na(longitude))
+
+# Leaflet default expect WGS84 (crs 4326), need custom CRS for HK1980 Grid (crs 2326)
+# https://rstudio.github.io/leaflet/projections.html
+hk_accidents_valid_sf <- st_as_sf(hk_accidents_valid, coords = c("longitude", "latitude"), crs = 4326, remove = FALSE)
+
+# Need to convert to POSIXct again, otherwise reactive filtering does not work
+# TODO: investigate why
+hk_accidents_valid_sf$Date <- as.Date(hk_accidents_valid_sf$Date, format = "%Y-%m-%d")
+
+CARTODB_POSITRON_TILE_URL <- "https://cartodb-basemaps-{s}.global.ssl.fastly.net/light_all/{z}/{x}/{y}.png"
+
+output$main_map <- renderLeaflet({
+  leaflet() %>%
+    addTiles(urlTemplate = CARTODB_POSITRON_TILE_URL) %>%
+    setView(114.2, 22.3, zoom = 12)
+})
+
+output$nrow_filtered <- reactive(nrow(filter_collision_data()))
+
+# Filter the collision data according to users' input
+filter_collision_data <- reactive({
+
+  data_filtered = dplyr::filter(hk_accidents_valid_sf, Date >= input$date_filter[1] & Date <= input$date_filter[2])
+
+  data_filtered = dplyr::filter(data_filtered,
+                                No__of_Casualties_Injured >= input$n_causality_filter[1] & No__of_Casualties_Injured <= input$n_causality_filter[2])
+
+  data_filtered = filter(data_filtered, Type_of_Collision %in% input$collision_type_filter)
+
+  data_filtered = filter(data_filtered, Severity %in% input$severity_filter)
```

Comment:
How come's `dplyr::` is specified for one set but not the other? The bottom two are using `dplyr::filter()` too, right? If not, then ignore my suggestion.

```suggestion
  data_filtered = filter(hk_accidents_valid_sf, Date >= input$date_filter[1] & Date <= input$date_filter[2])

  data_filtered = filter(data_filtered,
                                No__of_Casualties_Injured >= input$n_causality_filter[1] & No__of_Casualties_Injured <= input$n_causality_filter[2])

  data_filtered = filter(data_filtered, Type_of_Collision %in% input$collision_type_filter)

  data_filtered = filter(data_filtered, Severity %in% input$severity_filter)
```

Reply:
My vote will be to consistently apply the package + `::` prepend whenever there is a known namespace conflict, but open to discussing this. Also, not a deal breaker if we don't, but just stating my preference

Reply:
This is a good rule of thumb, and I'm in favour of this, just that I think sometimes loading `dplyr` does not throw up the warning about `dplyr::filter` conflicting with `stats:filter`. 😅